extern/BUILD.bazel

```diff
@@ -23,11 +25,8 @@ cc_library(
             ],
             "//conditions:default": [],
         }),
-    # This is needed to suppress warnings introduced by stb_image.h
-    copts = ["-Wno-sign-compare -Wno-unused-but-set-variable"],
-    visibility = [
-        "//core:__subpackages__",
-        "//examples:__subpackages__",
-        "//util:__subpackages__",
-    ],
+    linkopts = select({
+        "@bazel_tools//platforms:windows": ["-lsfml-window -lsfml-system -lopengl32 -lwinmm -lgdi32"],
+        "@bazel_tools//platforms:linux": ["-lsfml-window -lsfml-system -lGL"],
+    }),
```

Comment:
I'd prefer if we have "cascading" style of linkopts. That is, don't just put all link flags needed by the whole project in the base package, but put only the ones really needed, and then in dependent packages we don't have to repeat the ones that are already added in "parent" package, but just add any new ones needed for that particular package and so on.

Reply:
Yeah, I agree, but I am not a big fan of micro-managing with dependencies as it could get easily complicated and confusing (who is depending on whom? didn't I include it already? etc.)

Reply:
I'd like these dependencies moved to core package at least (I think util and extern shouldn't need any of these, and rest of the packages depend on core).